bears/vcs/git/GitCommitBear.py

```diff
@@ -110,6 +114,38 @@ def check_shortlog(self,
                     self,
                     "Shortlog of HEAD commit does not match given regex.")
 
+        if shortlog_imperative_check:
+            colon_pos = shortlog.find(':')
+            shortlog = shortlog[colon_pos:] if colon_pos != -1 else shortlog
+            has_flaws = self.check_imperative(shortlog)
+            if has_flaws:
+                bad_words = ', '.join([words[0] for words in has_flaws])
+                yield Result(self,
+                             "Shortlog of HEAD commit isn't imperative mood,"
+                             "bad words are '{}'".format(bad_words))
+
+    @staticmethod
+    def check_imperative(paragraph):
+        """
+        Check the given sentence/s for Imperatives.
+
+        :param paragraph: The input paragraph to be tested.
+        :returns: A list of invalid words.
+        """
```

Comment:
Lol. I am yet to push. :P
On 6 Apr 2016 03:14, "Lasse Schuirmann" <notifications@github.com> wrote:

> In bears/vcs/git/GitCommitBear.py
> <https://github.com/coala-analyzer/coala-bears/pull/315#discussion_r58618885>
> :
>
> > +            shortlog = shortlog[colon_pos:] if colon_pos != -1 else shortlog
> > +            has_flaws = self.check_imperative(shortlog)
> > +            if has_flaws:
> > +                bad_words = ', '.join([words[0] for words in has_flaws])
> > +                yield Result(self,
> > +                             "Shortlog of HEAD commit isn't imperative mood,"
> > +                             "bad words are '{}'".format(bad_words))
> > +
> > +    @staticmethod
> > +    def check_imperative(paragraph):
> > +        """
> > +        Check the given sentence/s for Imperatives.
> > +
> > +        :param paragraph: The input paragraph to be tested.
> > +        :returns: A list of invalid words.
> > +        """
>
> handling for non-present nltk data still missing also some of the other
> comments not yet addressed
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/coala-analyzer/coala-bears/pull/315/files/23d9af66bcb498b19adffe8effb5af1ec8aa4f40#r58618885>
>